src/main/rx/rx_unified_dsm.c

```diff
@@ -36,20 +45,22 @@ extern uint8_t ready_for_next_telemetry;
 #define USART usart_port_defs[serial_rx_port]
 #define SPECTRUM_BIND_PIN usart_port_defs[profile.serial.rx].rx_pin
 
-static uint8_t dsm_protocol = 0;
-#define DSMX_11_2048 0xb2
-#define DSMX_22_2048 0xa2
-#define DSM2_11_2048 0x12
-#define DSM2_22_1024 0x01
+uint8_t check_dsm_id (uint8_t id_byte){
+  if (id_byte == 0xb2) return DSMX_11_2048;
+  if (id_byte == 0xa2) return DSMX_22_2048;
+  if (id_byte == 0x12) return DSM2_11_2048;
+  if (id_byte == 0x01) return DSM2_22_1024;
+  return PROTOCOL_INVALID;
+}
```

Comment:
lets have the enum entries be the id values, that will remove the need for this function